core/src/main/scala/org/apache/spark/storage/OffHeapStore.scala

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.storage
+
+import java.nio.ByteBuffer
+import org.apache.spark.Logging
+import org.apache.spark.util.Utils
+
+import scala.util.control.NonFatal
+
+
+/**
+ * Stores BlockManager blocks on OffHeap.
+ * We capture any potential exception from underlying implementation
+ * and return with the expected failure value
+ */
+private[spark] class OffHeapStore(blockManager: BlockManager, executorId: String)
+  extends BlockStore(blockManager: BlockManager) with Logging {
+
+  lazy val offHeapManager: Option[OffHeapBlockManager] =
+    OffHeapBlockManager.create(blockManager, executorId)
+
+  logInfo("OffHeap started")
+
+  override def getSize(blockId: BlockId): Long = {
+    try {
+      offHeapManager.map(_.getSize(blockId)).getOrElse(0)
+    } catch {
+      case NonFatal(t) => logError(s"error in getSize from $blockId")
+        0
```

Comment:
From the code path, the existence of the block is always checked first. If contains(blockId) is false, the expected value is 0L. 

My understanding is that by returning 0L, it is the same as the store does not contain the block cache.